### ISSUE TITLE
Layers with different lattice vectors

### ISSUE THREAD

**Issue by Jelkf** [victorliu/S4#108]
Hi! I was wondering if there is a way for building layers with different lattice vectors or something like that

Thank you for any response!

**Comment by phoebe-p** [comment]
I don't believe there is a way to do this; there are two options I can think of:

1. Find the lowest common multiple of the lattice vector lengths (assuming both layers can be described using e.g. a square or rectangular unit cell) and define both layers in terms of this (of course it will make your unit cell more complicated, and may not be feasible if the unit cell gets too large)
2. If there is a medium separating the two layers with different lattice vectors, and that medium has a long optical path length compared to the vacuum wavelength of light you are interested in (i.e. you do not expect interference in this layer to be important), you could describe the angular redistribution of both surfaces in terms of a matrix (angular redistribution matrix methos). This is implemented in my package [RayFlare](https://rayflare.readthedocs.io/en/latest/index.html) which also interfaces with S4 through Python for RCWA functionality.

**Comment by Jelkf** [comment]
Ohh okay. 
Well, that interference is quite important. And the relation between the layers is not precisely square
So... I'll still looking for ways to solve this problem.

Thank you so much for your response, really appreciate it!
